src/streams.c

```diff
@@ -80,18 +80,20 @@ static Int READ_COMMAND(Obj *evalResult)
 
 /*
  * This function reads all code from the stream and returns either `fail` if the stream
- * cannot be opened, or a list of lists of length at most two, where the first entry is
- * either `true` or `false` depending on whether the statement was successfuly executed
- * and the second entry is the result of the statement if there was one.
+ * cannot be opened, or a list of lists of length three, where the first entry is
+ * either `true` or `false` depending on whether the statement was successfuly executed,
+ * the second entry is bound to the result of the statement if there was one, and unbound otherwise,
+ * and the third entry reflects whether the statement ended in a dual semicolon.
```

Comment:
This comment contradicts what the code does: It claims the lists inside the outer list always have length 3, but looking at the code, they can also have length 1. Indeed, your tests confirm this.